variables.tf

```diff
@@ -165,6 +165,12 @@ variable "performance_insights" {
   description = "Specifies whether Performance Insights is enabled or not"
 }
 
+variable "performance_insights_kms_key_id" {
+  type        = string
+  default     = ""
+  description = "The KMS key ID used for the PI encryption"
+}
```

Comment:
If you are sure that setting `performance_insights = false` isn't possible to solve this, then I propose to remove that variable and use this one to determine of the respective field on the resource should be `true` or `false`.

Reply:
Hmm... Thinking some more about this, I would like to suggest another solution. See the comment on `main.tf`...